plugin/pulpcore/plugin/stages/api.py

```diff
@@ -23,6 +23,56 @@ class Stage:
         """
         raise NotImplementedError(_('A plugin writer must implement this method'))
 
+    @staticmethod
+    async def batches(in_q, minsize=1):
+        """
+        Asynchronous iterator yielding batches of :class:`DeclarativeContent` from `in_q`. The
+        iterator will try to get as many instances of :class:`DeclarativeContent` as possible
+        without blocking, but at least `minsize` instances.
+
+        Args:
+            in_q (:class:`asyncio.Queue`): The queue to receive
+                :class:`~pulpcore.plugin.stages.DeclarativeContent` objects from.
+            minsize (int): The minimum batch size to yield (unless it is the final batch)
+
+        Yields:
+            A list of :class:`DeclarativeContent` instances
+
+        Examples:
+            Used in stages to get large chunks of declarative_content instances from
+            `in_q`::
+
+                class MyStage(Stage):
```

Comment:
What about using the >>> syntax here? I think that would give it python highlighting iirc. I use it here:

https://github.com/gmbnomis/pulp/blob/pipeline_customization/plugin/pulpcore/plugin/stages/declarative_version.py#L42-L56
Which renders like this:  https://docs.pulpproject.org/en/3.0/nightly/plugins/plugin-api/stages.html?highlight=myfirststage#declarativeversion

Reply:
This is a [literal block](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/basics.html#literal-blocks). `::` is the nicest thing in reST IMHO: No overhead for all practical purposes, looks completely natural in the source and gives nice Python highlighting in the docs (or whatever the default language is).

`>>>` is a [doctest block](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/basics.html#doctest-blocks) and serves a different purpose: "interactive Python sessions cut-and-pasted into docstrings"

Reply:
@gmbnomis Thank you for explaining this difference to me.